src/dataflow/src/sink/kafka.rs

```diff
@@ -571,6 +576,9 @@ where
 
                         match result {
                             Ok(()) => {
+                                assert!(write_frontier.borrow().less_equal(ts));
+                                write_frontier.borrow_mut().clear();
```

Comment:
this feels not great and maybe the most compelling reason to use a mutable antichain instead